typing/typetexp.ml

```diff
@@ -526,22 +526,22 @@ and transl_type_aux env ~row_context ~aliased ~policy styp =
   | Ptyp_alias(st, alias) ->
       let cty =
         try
-          let t = TyVarEnv.lookup_local ~row_context alias in
+          let t = TyVarEnv.lookup_local ~row_context alias.txt in
           let ty = transl_type env ~policy ~aliased:true ~row_context st in
           begin try unify_var env t ty.ctyp_type with Unify err ->
             let err = Errortrace.swap_unification_error err in
-            raise(Error(styp.ptyp_loc, env, Alias_type_mismatch err))
+            raise(Error(alias.loc, env, Alias_type_mismatch err))
           end;
           ty
         with Not_found ->
           let t, ty =
             with_local_level_if_principal begin fun () ->
               let t = newvar () in
-              TyVarEnv.remember_used alias t styp.ptyp_loc;
+              TyVarEnv.remember_used alias.txt t alias.loc;
```

Comment:
I propose to keep `styp.ptyp_loc` in this case: this would move the location from `Type_mismatch` in
```
let f (x: int as 'a) (y:float as 'a) = ...
```
to 
```ocaml
(y:float as 'a)
```
> Error: This type float should be an instance of type int

rather than just the alias, while only somehow degrading the quality of error messages in the pathological case
```ocaml
type t = 'a as 'b
```
(the location move from the `'b` alias to `'a as 'b` )

Reply:
Thanks, fixed.